dataplane/src/nat/iplist.rs

```diff
@@ -1,26 +1,91 @@
+///
+/// Consider for example the following two lists of IP prefixes:
+///
+/// ```text
+/// Initial set                               Target set
+/// +--------------------+                    +--------------------+
+/// |                    |                    |                    |
+/// | +----------------+ |                    | +----------------+ |
+/// | |    10.1.0.0/16 | |                    | | 192.168.1.0/24 | |
+/// | +----------------+ |                    | +----------------+ |
+/// |                    |  NAT must provide  |                    |
+/// | +----------------+ |   a mapping here   | +----------------+ |
+/// | |    10.0.5.0/24 | | -----------------> | |   10.10.0.0/16 | |
+/// | +----------------+ |                    | +----------------+ |
+/// |                    |                    |                    |
+/// | +----------------+ |                    | +----------------+ |
+/// | |    10.2.0.0/30 | |                    | |    10.8.0.0/31 | |
+/// | +----------------+ |                    | +----------------+ |
+/// |                    |                    |                    |
+/// |                    |                    | +----------------+ |
+/// |                    |                    | |    10.8.1.0/31 | |
+/// |                    |                    | +----------------+ |
+/// |                    |                    |                    |
+/// +--------------------+                    +--------------------+
+/// ```
+///
+/// Mapping an IP from the initial set to one from the target set is
+/// non-trivial, because we don't necessarily have a correspondence between the
+/// sizes of the prefixes in the two sets. To ease the mapping process, we
+/// introduce the [`IpList`] abstraction: it flattens the sets of endpoints into
+/// lists of IP addresses. The example above becomes:
+///
+/// ```text
+/// Initial set                                    Target set
+/// +---------------+                              +---------------+
+/// |               |                              |               |
+/// |      10.1.0.0 | 10.1.0.1 maps to 192.168.1.1 |   192.168.1.0 |
+/// |      10.1.0.1 | ---------------------------> |   192.168.1.1 |
+/// |         ...   |                              |         ...   |
+/// |  10.1.255.255 |                              | 192.168.1.255 |
+/// |      10.0.5.0 |                              |     10.10.0.0 |
+/// |         ...   |                              |         ...   |
+/// |    10.0.5.255 |                              | 10.10.255.255 |
+/// |      10.2.0.0 |                              |      10.8.0.0 |
+/// |      10.2.0.1 |  10.2.0.1 maps to 10.8.1.0   |      10.8.0.1 |
+/// |      10.2.0.2 | ---------------------------> |      10.8.1.0 |
+/// |      10.2.0.3 |                              |      10.8.1.1 |
+/// |               |                              |               |
+/// +---------------+                              +---------------+
```

Comment:
Interestingly, writing this piece of the doc made me realise that the mapping model is broken. By ignoring the subnet and broadcast addresses when establishing the mapping, we may end up with nonsensical translations such as this example, where we map to a network address.

This requires a fix that will come in its own Pull Request.